crates/swap_thorchain/src/provider.rs

```diff
@@ -0,0 +1,45 @@
+use crate::client::ThorchainSwapClient;
+use async_trait::async_trait;
+use primitives::{Chain, SwapQuote, SwapQuoteProtocolRequest};
+use swap_provider::{SwapError, SwapProvider};
+
+pub const PROVIDER_NAME: &str = "1inch";
+
+pub struct ThorchainProvider {
+    pub client: ThorchainSwapClient,
+}
+
+impl ThorchainProvider {
+    pub fn new(api_url: String, fee: f64, fee_referral_address: String) -> Self {
+        Self {
+            client: ThorchainSwapClient::new(api_url, fee, fee_referral_address),
+        }
+    }
+
+    pub fn new_box(api_url: String, fee: f64, fee_referral_address: String) -> Box<Self> {
```

Comment:
do we really need to have `new_box` method here? you can create box outside of this provider